prusti-tests/tests/verify/pass/generic/associated-types.rs

```diff
@@ -0,0 +1,33 @@
+use prusti_contracts::*;
+
+trait Model {
+  type Kind;
+}
+
+struct MyKind {}
+
+struct MyModel {}
+
+impl Model for MyModel {
+  type Kind = MyKind;
+}
+
+struct Guard<M: Model> {
+  kind: M::Kind,
+}
+
+// <MyModel as Model>::Kind is the same type as MyKind,
+// but it need to be normalized to see that.
+#[requires(gd.kind === gd.kind)]
+fn test1(gd: Guard<MyModel>){}
```

Comment:
What does this test? Is it that snapshot equality is not supported for an unresolved associated type, but it is supported for a concrete `struct` type? Could `MyModel` set `type Kind = i32` and then:

```suggestion
#[requires(gd.kind === x)]
fn test1(gd: Guard<MyModel>, x: i32) {}
```

Reply:
I pushed a new version that tests the same thing but makes more sense.